app/scripts-browserify/map.js

```diff
@@ -1,82 +1,151 @@
-          animation: google.maps.Animation.DROP,
-          icon: iconPath + 'currentLocation.png'
-        });
-        currentLocationMarker.setMap(map);
-
-        map.setCenter(currentLocation);
-      });
+    if (items.length === 0) {
+      return;
     }
 
+    let itemType = items[0].type;
+
     let markers = items.map(function(item) {
-      var colid = item.collection + '/' + item.id;
-      var boxText = document.createElement('div');
-
-      boxText.innerHTML = '<a href="/' + colid + '"><img src="/' + colid + '/thumbnail" width="220px" height="220px" /><h1>' + item.short_title + '</h1></a>';
-
-      var myOptions = {
-        content: boxText,
-        maxWidth: 220,
-        zIndex: null,
-        closeBoxURL: '',
-        alignBottom: true,
-        pixelOffset: new google.maps.Size(-110, -40),
+      if (!item.location || !item.location.lat || !item.location.lon) {
+        // Skip items without location.
+        return;
+      }
+
+      // Set the properties the markers share.
+      let markerData = {
+        position: { lat: item.location.lat, lng: item.location.lon }
       };
 
-      var marker = new google.maps.Marker({
-        position: { lat: item.latitude, lng: item.longitude },
-        title: item.short_title,
-        animation: google.maps.Animation.DROP,
-        icon: iconPath + 'pin.png',
-        item: item,
-        infobox: new InfoBox(myOptions),
-      });
+      // Hash markers are all going to be aggregated so all we need to inject
+      // is a count.
+      if(item.type === 'hash') {
+        markerData.subCount = item.count;
+      } else if (item.type === 'asset') {
+        var colid = item.assetData.collection + '/' + item.assetData.id;
+        var boxText = document.createElement('div');
+
+        boxText.innerHTML = '<a href="/' + colid + '"><img data-src="/' + colid + '/thumbnail" width="220px" height="220px" /><h1>' + item.assetData.short_title + '</h1></a>';
+
+        var infoboxConfiguration = {
+          content: boxText,
+          maxWidth: 220,
+          zIndex: null,
+          closeBoxURL: '',
+          alignBottom: true,
+          pixelOffset: new google.maps.Size(-110, -40),
+        };
 
-      return marker;
+        markerData.title = item.assetData.short_title;
+        markerData.icon = iconPath + 'pin.png';
+        markerData.infobox = new InfoBox(infoboxConfiguration);
+      }
+
+      return new google.maps.Marker(markerData);
     });
 
-    markers.forEach(marker => {
-      marker.addListener('mousedown', function() {
-        var currentMarker = this;
-        $.each(markers, function(i, marker) {
-          if(marker !== currentMarker) {
-            marker.infobox.close();
-          }
+    // Filter away markers without location.
+    markers = markers.filter(Boolean);
+
+    if (itemType === 'asset') {
+      let googleMapReference = this.googleMap;
+      markers.forEach(marker => {
+        marker.addListener('mousedown', function() {
```

Comment:
Great! Made the mistake of using `click` which to my surprise doesn't work on touch events.

Reply:
I might have stolen it off of your branch ;)